std/traits.d

```diff
@@ -8590,3 +8591,85 @@ enum isCopyable(S) = is(typeof(
     static assert(isCopyable!int);
     static assert(isCopyable!(int[]));
 }
+
+/++
+ + Determines if `F` is a strongly `pure` function. This is true when :
+ + $(UL
+ +   $(LI Each parameter is `immutable`, or can be converted automatically to `immutable`
+ +   (i.e. has no mutable indirections) and is passed by value)
+ +   $(LI The Return Type is `immutable` or can be impicitly converted to `immutable`)
+ + )
+ + Params:
+ + F = The function to be tested
+ +
+ + Returns:
+ + `true` if the function is strongly `pure`, `false` if otherwise.
+ + ++/
```

Comment:
No leading stars or pluses please.
Have a look at https://dlang.org/dstyle.html#phobos_documentation (i.e. of sections indentation is also nice)